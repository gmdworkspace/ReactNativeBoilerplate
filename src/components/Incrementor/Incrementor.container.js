import { connect } from 'react-redux';
import { incrementCount } from '../../actions/creators';

const mapStateToProps = function(state) {
  return {
    count : state.app.count
  };
};

const mapDispatchToProps = function(dispatch) {
  return {
    incrementCount : () => dispatch(incrementCount())
  };
};

export default connect(mapStateToProps, mapDispatchToProps);