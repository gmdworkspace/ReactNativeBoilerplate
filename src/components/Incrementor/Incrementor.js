import React, {Component} from 'react';
import {Platform, StyleSheet, Text, View, Button} from 'react-native';
import PropTypes from 'prop-types';

type Props = {};
class Incrementor extends Component<Props> {

  constructor() {
    super();
    this.handleOnPress = this.handleOnPress.bind(this);
  }

  handleOnPress() {
    const { incrementCount } = this.props;
    incrementCount();
  }

  render() {
    const { count } = this.props;
    return (
      <View>
        <Text style={styles.welcome}>
          Hello count:{count}
        </Text>

        <Button
          onPress={this.handleOnPress}
          title="Press Me"
          color="#841584"
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  }
});

Incrementor.propTypes = {
  count: PropTypes.number
};

export default Incrementor;
