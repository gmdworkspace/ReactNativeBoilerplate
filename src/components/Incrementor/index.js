import Incrementor from './Incrementor';
import container from './Incrementor.container';

export default container(Incrementor);