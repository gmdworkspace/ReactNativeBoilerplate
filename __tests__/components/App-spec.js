import React from 'react';
import Incrementor from '../../src/components/Incrementor/Incrementor';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const component = renderer.create(<Incrementor />).toJSON();
  expect(component).toMatchSnapshot();
});