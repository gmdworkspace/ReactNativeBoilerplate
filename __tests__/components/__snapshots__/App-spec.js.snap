// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }
  >
    Hello count:
  </Text>
  <View
    accessibilityComponentType="button"
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#841584",
            },
          ]
        }
      >
        Press Me
      </Text>
    </View>
  </View>
</View>
`;
